assume/strategies/learning_advanced_orders.py

```diff
@@ -264,17 +265,34 @@ def create_observation(
         # =============================================================================
         # 1.1 Get the Observations, which are the basis of the action decision
         # =============================================================================
-        scaling_factor_res_load = self.max_demand
 
+        # defines bounds of observation space
+        # stays here as it is unit specific, and different forecasts might apply for different units
+        # different handling would require an extra unit loop at learning role intiliazation and unit specific max/min values
+        # further forecasts might change during the simulation if advanced forecasting is used
+        self.max_market_price = max(unit.forecaster[f"price_{market_id}"])
```

Comment:
this can be done in __init__ so it is done only once and not during every call of the function

Reply:
I disagree for the reasons stated in the lengthly comment

Reply:
@kim-mskw what lengthy comment? Don't see it

Reply:
I argue that the market price max might be different for different units in the future, hence it should be unit specific as well 

# stays here as it is unit specific, and different forecasts might apply for different units
# different handling would require an extra unit loop at learning role intiliazation and unit specific max/min values
# further forecasts might change during the simulation if advanced forecasting is used

Reply:
For the sake of performance, we decided to sacrifice the accommodation of future developments. Hence, we move the scaling up to the init, where it is only called once.